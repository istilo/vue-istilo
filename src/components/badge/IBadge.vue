<script setup lang="ts">
const props = defineProps<{
	/**
	 * A number that will show in the badge
	 */
	count: number;
}>();
</script>

<template>
	<span class="i-badge">
		{{ props.count }}
	</span>
</template>
