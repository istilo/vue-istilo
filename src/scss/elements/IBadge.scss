@import '@/scss/helpers';
@import '@/scss/functions';
@import '@/scss/mixins';

// ============================================================
// Name:            Badge
// Description:     Component to create notification badges
//
// Component:       `#{$prefix}badge`
//
// ============================================================

// Variables
// ============================================================

$badge-size: 22px !default;

$badge-padding-vertical: 4px !default;
$badge-padding-horizontal: 10px !default;
$badge-border-radius: 500px !default;
$badge-background: theme-color(primary) !default;
$badge-color: color(inverse) !default;
$badge-font-size: font-size(small);

$badge-hover-color: color(inverse) !default;

/* ============================================================
	Component: Badge
   ============================================================ */

/*
 * 1. Style
 * 2. Center child vertically and horizontally
 */

.#{$prefix}badge {
	box-sizing: border-box;
	min-width: $badge-size;
	height: $badge-size;
	padding: $badge-padding-vertical $badge-padding-horizontal;
	border-radius: $badge-border-radius;
	vertical-align: middle;
	margin-left: 8px;

	/* 1 */
	background: $badge-background;
	color: $badge-color;
	font-size: $badge-font-size;

	/* 2 */
	display: inline-flex;
	justify-content: center;
	align-items: center;
}

/*
 * Required for `a`
 */

.#{$prefix}badge:hover,
.#{$prefix}badge:focus {
	color: $badge-hover-color;
	text-decoration: none;
	outline: none;
}
