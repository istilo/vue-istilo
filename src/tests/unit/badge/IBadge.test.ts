import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { shallowMount, VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import IBadge from '@/components/badge/IBadge.vue';

enableAutoUnmount(afterEach);

describe('Istilo Badge', () => {
	let wrapper: VueWrapper<any>;

	beforeEach(() => {
		wrapper = shallowMount(IBadge, {
			propsData: {
				count: 0,
			},
		});

		expect(wrapper.exists()).toBe(true);
	});
	afterEach(() => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Should render correctly', async () => {
		expect(wrapper.element.tagName).toBe('SPAN');
		expect(wrapper.classes()).toContain('i-badge');
		expect(wrapper.text()).toBe('0');
	});

	it('Should props.count equals to 99', async () => {
		await wrapper.setProps({ count: 99 });

		expect(wrapper.text()).toBe('99');
	});
});
