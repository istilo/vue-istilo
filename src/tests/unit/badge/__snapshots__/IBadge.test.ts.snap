// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Istilo Badge > Should props.count equals to 99 1`] = `"<span class=\\"i-badge\\">99</span>"`;

exports[`Istilo Badge > Should render correctly 1`] = `"<span class=\\"i-badge\\">0</span>"`;
